ros2_control_demo_hardware/CMakeLists.txt

```
@@ -7,7 +7,7 @@ if(NOT CMAKE_CXX_STANDARD)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  add_compile_options(-Wall -Wextra)
+  add_compile_options(-Wall -Wextra -Werror -Wpedantic -Wshadow)
```

Let's not add additional flags which are not using anywhere else.

```suggestion
  add_compile_options(-Wall -Wextra)
```